determined-ai/determined
# chore: Implement Theming updates in Determined [WEB-1726]

## Description

Sibling PR for: https://github.com/determined-ai/hew/pull/10

This PR implements the new `UIProvider` and other Theme related utilities that were introduced in the linked sibling PR. 
The updates were largely accomplished by implementing the original `UIProvider` as a new `ThemeProvider` and adding new `theme` and `isDarkMode` logic to the current `useUI` hook. 

~After this PR is merged in: https://github.com/determined-ai/hew/pull/39 I will update the Hew version in hew and this repo.~





## Test Plan
1. Log in to the cluster and ensure you are on the dashboard page.
2. Using the Theme toggle which can be accessed by clicking on your username in the top left corner of the page, switch the theme between System, Light, and Dark. Ensure the theme changes accordingly. 
3. Go to the Settings menu by clicking the "settings" item which can also be accessed by clicking on your username in the top left corner of the page.
4. Change the "Theme Mode" between System, Light, and Dark. Ensure the theme changes accordingly. 
5. Change the theme to "Dark" and logout, ensure that when you are logged out the theme continues to be dark.
6. Log in, ensure when you are logged in the theme continues to be dark.
7. Change the theme to "Light" and logout, ensure when you are logged out the theme continues to be light.
6. Log in, ensure when you are logged in the theme continues to be light.





## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket
WEB-1726
